pkg/registry/core/service/storage/rest_test.go

```diff
@@ -1561,273 +1066,186 @@ func TestServiceRegistryUpdateMultiPortExternalService(t *testing.T) {
 
 func TestServiceRegistryGet(t *testing.T) {
 	ctx := genericapirequest.NewDefaultContext()
-	storage, registry, server := NewTestREST(t, nil, singleStackIPv4)
+	storage, server := NewTestREST(t, singleStackIPv4)
 	defer server.Terminate(t)
-	registry.Create(ctx, &api.Service{
-		ObjectMeta: metav1.ObjectMeta{Name: "foo"},
-		Spec: api.ServiceSpec{
-			Selector: map[string]string{"bar": "baz"},
-		},
-	}, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-	storage.Get(ctx, "foo", &metav1.GetOptions{})
-	if e, a := "foo", registry.GottenID; e != a {
+	_, err := storage.Create(ctx, svctest.MakeService("foo"), rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating service: %v", err)
+	}
+	obj, _ := storage.Get(ctx, "foo", &metav1.GetOptions{})
+	svc := obj.(*api.Service)
+	if e, a := "foo", svc.Name; e != a {
```

Comment:
this is a funky if statement. valid but funky.

Reply:
I usually see it as `want, got = expr, expr; got != want` but this is valid so I won't change it now.  As tests move to the single-layer we can scrub the hell out of them